src/project.rs

```diff
@@ -187,17 +189,39 @@ impl FuzzProject {
             rustflags.push_str(" -Cinstrument-coverage");
         }
 
-        match build.sanitizer {
-            Sanitizer::None => {}
-            Sanitizer::Memory => {
-                // Memory sanitizer requires more flags to function than others:
-                // https://doc.rust-lang.org/unstable-book/compiler-flags/sanitizer.html#memorysanitizer
-                rustflags.push_str(" -Zsanitizer=memory -Zsanitizer-memory-track-origins")
+        if !matches!(build.sanitizer, Sanitizer::None) {
+            // Select the appropriate sanitizer flag for the given rustc version:
+            // either -Zsanitizer or -Csanitizer.
+            let rust_version_string = rust_version_string()?;
+            let rust_version =
+                RustVersion::from_str(&rust_version_string).map_err(|e| anyhow::anyhow!(e))?;
```

Comment:
Nitpick: seems like these two steps could be collapsed into a single, fallible `RustVersion` constructor:

```rust
impl RustVersion {
    pub fn discover() -> anyhow::Result<Self> {
        let version_string = rust_version_string()?;
        let me = Self::from_str(&version_string)?;
        Ok(me)
    }
}
```

No need to make API-consumers do two things when they could do one; just an opportunity for them to mess things up in between or not connect the two interfaces properly.

Reply:
The reason these are two separate steps is that we need the version string too, to detect whether the compiler is nightly or not.

I considered adding the version string to the `RustVersion` struct, but that would make it no longer `Copy`, and neither API looked obviously better than the other.

If you want me to encapsulate everything, I guess I could turn `RustVersion` into this:

```rust
pub struct RustVersion {
    pub major: u32,
    pub minor: u32,
    // we don't care about the patch version and it's a bit of a pain to parse
    nightly: bool,
}
```

and with the nightly flag being resolved in `RustVersion` we don't need to expose the string outside the module at all.